pkg/api/serialization_test.go

```diff
@@ -174,7 +174,8 @@ func TestCommonKindsRegistered(t *testing.T) {
 				t.Error(err)
 			}
 			defaults := gv.WithKind("")
-			if _, got, err := api.Codecs.LegacyCodec().Decode([]byte(`{"kind":"`+kind+`"}`), &defaults, nil); err != nil || gvk != *got {
+			var got *schema.GroupVersionKind
+			if obj, got, err = api.Codecs.LegacyCodec().Decode([]byte(`{"kind":"`+kind+`"}`), &defaults, obj); err != nil || gvk != *got {
```

Comment:
@smarterclayton @deads2k without the `into` object, we get an internal type by default. According to https://github.com/kubernetes/kubernetes/blob/6a4d5cd7cc58e28c20ca133dab7b0e9e56192fe3/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go#L96-L96 this is intentional. Is that intent wrong or this test?